### PR TITLE
QC subworkflow implementation iteration-1

### PR DESCRIPTION
**Pull request by abhi18av**
TODOs:

- [x] Move the `templates` to the top-level `bin` folder
- [ ] Implement the QC workflow discussed in https://github.com/nf-core/gwas/issues/29

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/gwas/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/gwas _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: subworkflows/local/qc.nf [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+      .multiMap {  it ->
+         raw_ch: it
+         bim_ch: it[1]
+         inpmd5ch : it
+       }.set {checked_input}
+
+}
+
+
+workflow QC_PROCESSES {
```
Comment (abhi18av) [inline subworkflows/local/qc.nf:128]: @Mxrcon , I've tried to simplify the overall implementation design and have centered it around the 4 channels such as `qc1_ch` ... `qc4_ch`.

I request you to please have a look at the `QC_PROCESSES` workflow and compare it with the DAG here https://github.com/nf-core/gwas/issues/29#issuecomment-1160037132 ?


The code within the `QC_PROCESSES` itself is pseudo-Nextflow at the moment and relies on the `pipes` concept.

I have a feeling that there must be a few pathways which I might have missed but would appreciate a quick review from you to make sure that there isn't anything obvious I am missing out, before taking it forward.

Comment (Mxrcon) [reply subworkflows/local/qc.nf:128]: @abhi18av This workflow seems to be going in the right direction, at least I felt that every pathway is inserted on it. maybe we can double check with sanity tests after a stub run, what do you think? Also what are the next steps on the workflow deployment? how can I be useful?

**Review by Mxrcon — COMMENTED** [review]
This PR seems to be in a good state, addresses the first steps that we discussed as relevant on the first steps of the qc-wf development. The next step seems to be the usage of stubs and concrete testing of input/output channels. Also, I'm available to help with anything you need, please if necessary tell me where I can be useful.

**Comment by abhi18av** [comment]
Merging this PR as it getting too big to be useful for reviews or checking diffs.

The current status is 
- [x] All modules have been transformed in DSL-2
- [x] The profiles have been updated to facilitate the development 
- [x] We're now able to run the base pipeline on the cluster


```
 E X T F L O W  ~  version 22.08.0-edge
Launching `main.nf` [boring_church] DSL2 - revision: acea032e39


------------------------------------------------------
                                        ,--./,-.
        ___     __   __   __   ___     /,-._.--~'
  |\ | |__  __ /  ` /  \ |__) |__         }  {
  | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                        `._,._,'
  nf-core/gwas v1.0dev
------------------------------------------------------
Core Nextflow options
  runName        : boring_church
  containerEngine: singularity
  container      : quay.io/h3abionet_org/py3plink
  launchDir      : /home/cust100107_vol1/abhi18av/gwas
  workDir        : /home/cust100107_vol1/abhi18av/gwas/work
  projectDir     : /home/cust100107_vol1/abhi18av/gwas
  userName       : abhi18av
  profile        : debug,singularity,qc_wf
  configFiles    : /home/cust100107_vol1/abhi18av/gwas/nextflow.config, /home/cust100107_vol1/abhi18av/gwas/params_1000G_h3abionet.config

!! Only displaying parameters that differ from the pipeline defaults !!
------------------------------------------------------
If you use nf-core/gwas for your analysis please cite:

* The nf-core framework
  https://doi.org/10.1038/s41587-020-0439-x

* Software dependencies
  https://github.com/nf-core/gwas/blob/master/CITATIONS.md
------------------------------------------------------
The batch file is 0
Sexinfo available command
executor >  slurm (1)
[54/2e6752] process > NFCORE_GWAS:GWAS:QC_WF:QC_INPUT_VALIDATION:NO_SAMPLESHEET [100%] 1 of 1 ✔
-[nf-core/gwas] Pipeline completed successfully-


```

@shaze @jeantristanb

File: subworkflows/local/qc.nf (hunk 1 above)
Comment (abhi18av) [reply subworkflows/local/qc.nf:324]: @Mxrcon , I think that now we've reached a stage where we can effectively test our changes https://github.com/nf-core/gwas/pull/32#issuecomment-1204796864 

I'll share my dev workflow with you when we connect next and then we can continue working w.r.t the templates and finalize the QC workflow.

**Pull request merged**